HAST-314: ⭐ Hast.Samples.Consumer GUI bug fixes and documentation housekeeping

sarahelsaig opened the pull request:
[HAST-314](https://lombiq.atlassian.net/browse/HAST-314)

[HAST-314]: https://lombiq.atlassian.net/browse/HAST-314?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sarahelsaig committed:
Custom configuration handle JSON.

sarahelsaig committed:
fix timing report file name for Azure Alveo U250 driver

sarahelsaig committed:
Merge remote-tracking branch 'origin/issue/HAST-175' into issue/HAST-314

sarahelsaig committed:
Merge remote-tracking branch 'origin/issue/HAST-175' into issue/HAST-314

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/HAST-314

sarahelsaig committed:
Fix typos.

sarahelsaig committed:
Update broken URLs.

sarahelsaig committed:
Update .NET Core 3.1 references.

sarahelsaig committed:
Add missing Azure link and Nimbix outdated warning.

Piedone commented on src/Hastlayer/Hast.Vitis/Docs/Nimbix.md:
```diff
@@ -1,6 +1,8 @@
 # Nimbix usage
 
 
+> ⚠️ The instructions below are outdated, as the mentioned Alveo trial is no longer available. Some of the names have changed since Nibix has been acquired by Atos, and they only seem to offer enterprise solutions now. The rest of the document is retained in case it may be a useful reference for anyone who wants to use their services with Hastlayer.
```
```suggestion
> ⚠️ The instructions below are outdated, as the mentioned Alveo trial is no longer available. Some of the names have changed since Nimbix has been acquired by Atos, and they only seem to offer enterprise solutions now. The rest of the document is retained in case it may be a useful reference for anyone who wants to use their services with Hastlayer.
```

Piedone commented on src/Hastlayer/Hast.Vitis/Readme.md:
```diff
@@ -20,7 +18,8 @@ For Nimbix-specific instructions see [the Nimbix docs](Docs/Nimbix.md).
 Even after everything is installed, you have to make sure that the executing user's environment variables are correctly set by sourcing the setup scripts [as described in the documentation](https://docs.xilinx.com/r/en-US/ug1400-vitis-embedded/Setting-Up-the-Environment-to-Run-the-Vitis-Software-Platform). You can add these commands into the `~/.bashrc` file to avoid having to type them every tim ...
 
-For setup instructions on the Nimbix cloud see the [Nimbix-specific instructions](Docs/Nimbix.md).
+For Azure-specific instructions see [the Azure NP docs](Docs/AzureReadme.md).
+For Nimbix-specific instructions see [the Nimbix docs](Docs/Nimbix.md).
```
```suggestion
- For Azure-specific instructions see [the Azure NP docs](Docs/AzureReadme.md).
- For Nimbix-specific instructions see [the Nimbix docs](Docs/Nimbix.md).
```

sarahelsaig committed:
Update src/Hastlayer/Hast.Vitis/Docs/Nimbix.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

sarahelsaig committed:
Update src/Hastlayer/Hast.Vitis/Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

sarahelsaig committed:
update submodules

sarahelsaig committed:
Merge branch 'issue/HAST-314' of https://github.com/Lombiq/Hastlayer-SDK into issue/HAST-314

sarahelsaig committed:
eliminate package downgrade

sarahelsaig committed:
Add ErrorOnDuplicatePublishOutputFiles

to avoid "error NETSDK1152: Found multiple publish output files with the same relative path".

sarahelsaig committed:
Fix bug where Azure composer is not used for AzureAlveoU250Driver.

sarahelsaig committed:
Fix "The xclbin DCP file is not a netlist." bug.

sarahelsaig committed:
Restore missing driver config.

sarahelsaig committed:
Fix crash if compilation fails the first time and passes the second time.

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/HAST-314

sarahelsaig committed:
Include SingleBinaryPath in Hast.NuGet.Console.

sarahelsaig committed:
Only use xclbin files.

sarahelsaig committed:
Update NuGet packages.

sarahelsaig committed:
Add cross-compilation instructions to Hast.NuGet.Console.

sarahelsaig committed:
Fix package reference names.

Piedone commented on Directory.Build.props:
```diff
@@ -1,6 +1,10 @@
   <Import Project="tools/Lombiq.Analyzers/Lombiq.Analyzers/Build.props" />
 
+  <PropertyGroup>
+    <ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles>
```
Comment why this is necessary.

Piedone committed:
Typo

sarahelsaig committed:
Add comment why ErrorOnDuplicatePublishOutputFiles has to be disabled.

Piedone committed:
Typo

[skip ci]

Piedone committed:
Updating submodules to merged dev

[skip ci]
